### PR TITLE
Add Coherence related code change and docs

### PR DESCRIPTION
**Pull request by zhengchang907**
* Add a sample Coherence application with one cache defined to /clusterwithcoherence/coherence/gar
* Add a sample EAR definition to allow WebLogic-Cafe to use the Coherence app
* Duplicate WebLogic-Cafe app under /clusterwithcoherence and addCafeCacheServlet with logic interact with defined Coherence cache
* Add documentation to explain the steps of playing with cache
* Remove `Security` directory
* Remove ELK section, NSG rules configuration of Cluster, and add `Create User-Assigned Managed Identity` section

### CODE REVIEW

File: clusterwithcoherence/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Deploying a Java EE Application with Coherence on Azure Using a WebLogic Virtual Machine Cluster
+This demo shows how you can deploy a Java EE application with Coherence to Azure into a WebLogic cluster using a marketplace solution.
+## Setup
+
+* Install the Window Subsystem for Linux (or other Linux, we used [WSL 2 with Ubuntu 18.04 LST](https://docs.microsoft.com/en-us/windows/wsl/install-win10)).
+* Install Maven, zip, jar in your WSL
+* Download this repository somewhere in your WSL file system (easiest way might be to download as a zip and extract).
```
Comment (galiacheng) [inline clusterwithcoherence/README.md:7]: "easiest way might be to download as a zip and extract" -> I would suggest to also add the commands to download and extract the repo.

Comment (galiacheng) [reply clusterwithcoherence/README.md:7]: The same with cluster page

File: clusterwithcoherence/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+* In "Coherence", toggle "Use Coherence cache?" to `Yes`. Leave other values to default.
+* Click Next:Review + create
+   * On the Summary blade you must see "Validation passed".  If you don't see this, you must troubleshoot and resolve the reason.  After you have done so, you can continue.
+   * On the final screen, click Create.
+* It will take some time for the WebLogic cluster to properly deploy (could be up to an hour). Once the deployment completes, in the portal go to 'All resources'.
+* Find and click on adminVM. Copy the DNS name for the admin server. You should be able to log onto http://`<admin server DNS name>`:7001/console successfully using the credentials above.  If you are not able to log in, you must troubleshoot and resolve the reason why before continuing.
+
+## Prepare Coherence App
+* Open a command line window and cd to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/gar`.
+* Run "jar cvf ../ear/weblogic-cafe-coherence.gar *", this will generate a weblogic-cafe-coherence.gar file to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/ear`.
```
Comment (galiacheng) [inline clusterwithcoherence/README.md]: Run "jar cvf ../ear/weblogic-cafe-coherence.gar *" -> Run `jar cvf ../ear/weblogic-cafe-coherence.gar *`

File: clusterwithcoherence/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+## Prepare Enterprise App combined WebLogic-Cafe and Coherence App
+* Open a command line window and cd to  `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/weblogic-cafe`
+* Run "mvn clean install"
+* Cd to `/target`
+* Copy generated .war file to `../weblogic-on-azure/clusterwithcoherence/coherence/ear` by running `cp weblogic-cafe.war ../../coherence/ear/`
+* Cd to `../weblogic-on-azure/clusterwithcoherence/coherence/ear`
+* Run "jar -cvf weblogic-cafe-pack.ear weblogic-cafe-coherence.gar weblogic-cafe.war META-INF/", this will generate a weblogic-cafe-pack.ear file.
```
Comment (galiacheng) [inline clusterwithcoherence/README.md]: Run "jar -cvf weblogic-cafe-pack.ear weblogic-cafe-coherence.gar weblogic-cafe.war META-INF/" -> Run `jar -cvf weblogic-cafe-pack.ear weblogic-cafe-coherence.gar weblogic-cafe.war META-INF/`

File: clusterwithcoherence/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+* It will take some time for the WebLogic cluster to properly deploy (could be up to an hour). Once the deployment completes, in the portal go to 'All resources'.
+* Find and click on adminVM. Copy the DNS name for the admin server. You should be able to log onto http://`<admin server DNS name>`:7001/console successfully using the credentials above.  If you are not able to log in, you must troubleshoot and resolve the reason why before continuing.
+
+## Prepare Coherence App
+* Open a command line window and cd to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/gar`.
+* Run "jar cvf ../ear/weblogic-cafe-coherence.gar *", this will generate a weblogic-cafe-coherence.gar file to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/ear`.
+
+## Prepare Enterprise App combined WebLogic-Cafe and Coherence App
+* Open a command line window and cd to  `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/weblogic-cafe`
+* Run "mvn clean install"
```
Comment (galiacheng) [inline clusterwithcoherence/README.md]: Run "mvn clean install" -> Run `mvn clean install`

File: clusterwithcoherence/README.md (hunk 2 above)
Comment (galiacheng) [reply clusterwithcoherence/README.md]: Can we use `mvn build` to generate the `gar` file?

Comment (galiacheng) [reply clusterwithcoherence/README.md]: Please refer to [Building a Real Application with Maven](https://docs.oracle.com/middleware/1221/core/MAVEN/real_app.htm#MAVEN8918)

**Comment by zhengchang907** [comment]
Close current PR, will create a new one to adopt latest changes

**Pull request closed without merging**